target_chains/cosmwasm/tools/src/helper.ts

```diff
@@ -0,0 +1,113 @@
+import { readFileSync, existsSync, mkdirSync } from "fs";
+import { rimrafSync } from "rimraf";
+import AdmZip from "adm-zip";
+import path from "path";
+import { DownloaderHelper } from "node-downloader-helper";
+
+// We have released the compile contacts on github. If a chain needs some specific
+// feature in a contract, a version of the contract with that specific features is
+// also released. For example, "injective.zip" for injective.
+// Given a contract version (eg 1.2.0) and  zipFileNames (eg ["injective", "osmosis"])
+// this method will return an object with key = zipFileName and value = compiledCode
+export async function getContractBytesDict(
+  artifactZipFileNames: string[],
+  version: string
+) {
+  const githubArtifactsLink = `https://github.com/pyth-network/pyth-crosschain/releases/download/pyth-cosmwasm-contract-v${version}/`;
+  const tmpCodeStorageDir = "./tmp";
+  // clear tmp directory before downloading contracts
+  rimrafSync(tmpCodeStorageDir);
+
+  if (existsSync(tmpCodeStorageDir) === false) {
+    mkdirSync(tmpCodeStorageDir, { recursive: true });
+  }
+
+  const uniqueArtifactsZipName = Array.from(new Set(artifactZipFileNames));
+
+  // download zip files
+  await Promise.all(
+    uniqueArtifactsZipName.map(async (artifactZipName) => {
+      return new Promise<void>((resolve, reject) => {
+        const dl = new DownloaderHelper(
+          githubArtifactsLink + artifactZipName + ".zip",
+          tmpCodeStorageDir
+        );
+
+        dl.on("end", () => {
+          console.log("Download Completed");
+          resolve();
+        });
+        dl.on("error", (err) => {
+          console.log("Download Failed", err);
+          reject(err);
+        });
+        dl.start().catch((err) => {
+          console.error(err);
+          reject(err);
+        });
+      });
+    })
+  );
+
+  // extract zip files
+  uniqueArtifactsZipName.map(async (artifactZipName) => {
+    const zip = new AdmZip(
+      path.resolve(tmpCodeStorageDir + "/" + artifactZipName + ".zip")
+    );
+    zip.extractAllTo(path.resolve(tmpCodeStorageDir));
+  });
+
+  let contractBytesDict: { [fileName: string]: Buffer } = {};
+  for (let uniqueArtifactZipName of uniqueArtifactsZipName) {
+    const contractBytes = readFileSync(
+      tmpCodeStorageDir + "/" + uniqueArtifactZipName + "/pyth_cosmwasm.wasm"
+    );
+    contractBytesDict[uniqueArtifactZipName] = contractBytes;
+  }
+
+  // clear tmp directory after downloading contracts
+  rimrafSync(tmpCodeStorageDir);
+
+  return contractBytesDict;
+}
+
+export const WORMHOLE_CONTRACT_VERSION = "2.14.9";
+
+// This method returns the file name which stores the deployment result for
+// wormhole stub
+export function getWormholeFileName(
+  chainId: string,
+  version: string,
+  mainnet: boolean
+): string {
+  const WORMHOLE_STORAGE_DIR = mainnet
+    ? "../wormhole-stub/mainnet"
+    : "../wormhole-stub/testnet";
```

Comment:
extract constant variables to hold these string constants. that way you can rename the directories in code without chasing down all of the instances of the string "wormhole-stub"

Reply:
also why is this filepath "wormhole-stub/testnet" and the pyth one is "testnet/instantiate-pyth"? seems like those should be in the same order

Reply:
also, might be nice to write down what filepaths this generates and what gets stored there. Could be a good code comment on wherever you put the string constants.

Reply:
Extracting constant variables. This is the only place it is being used and hence the only place we would need to make change.